src/ui/Font.hpp

```diff
@@ -16,26 +16,51 @@ class Font {
 	Font& operator=(const Font&) = delete;
 	Font& operator=(Font&&) = delete;
 
+	//! helper function for sizeText(), except operates on just one line.
+	int sizeLine(std::string str, int* out_w, int* out_h, int override_pointsize = 0) const;
+
+	//! helper function for drawTextColor(), except operates on just one line.
+	void drawLineColor(std::string str, int x, int y, const Uint32& color, int override_pointsize);
+
 public:
 	//! built-in font
 	static const char* defaultFont;
 
+	//! tab-width in spaces
+	static const int TAB_WIDTH = 4;
+	//! The following is basically a prerendered tab replacement...must be maintained to match tab_width. //TODO: Why even have TAB_WIDTH, then?
```

Comment:
The TODO here is a lie, we need TAB_WIDTH instead if we want true tab behavior...need to fix this, tab currently just always inserts `TAB_WIDTH` number of spaces, which is not the correct behavior. It should align up to the next multiple of `TAB_WIDTH`